Sources/IO/Misc/OBJWriter/index.d.ts

```diff
@@ -0,0 +1,52 @@
+import vtkPolyData from "../../../Common/DataModel/PolyData";
+import { vtkAlgorithm, vtkObject } from "../../../interfaces";
+
+/**
+ * 
+ */
+export interface IOBJWriterInitialValues {}
+
+type vtkOBJWriterBase = vtkObject & vtkAlgorithm;
+
+export interface vtkOBJWriter extends vtkOBJWriterBase {
+
+	/**
+	 *
+	 * @param inData 
+	 * @param outData 
+	 */
+	requestData(inData: any, outData: any): void;
+}
+
+/**
+ * Method used to decorate a given object (publicAPI+model) with vtkOBJWriter characteristics.
+ *
+ * @param publicAPI object on which methods will be bounds (public)
+ * @param model object on which data structure will be bounds (protected)
+ * @param {IOBJWriterInitialValues} [initialValues] (default: {})
+ */
+export function extend(publicAPI: object, model: object, initialValues?: IOBJWriterInitialValues): void;
+
+/**
+ * Method used to create a new instance of vtkOBJWriter
+ * @param {IOBJWriterInitialValues} [initialValues] for pre-setting some of its content
+ */
+export function newInstance(initialValues?: IOBJWriterInitialValues): vtkOBJWriter;
+
+/**
+ * 
+ * @param {vktPolyData} polyData 
+ */
+export function writeOBJ(polyData: vtkPolyData): vtkPolyData;
+
+/**
+ * vtkOBJWriter writes wavefront obj (.obj) files in ASCII form. OBJ files
+ * contain the geometry including lines, triangles and polygons. Normals and
+ * texture coordinates on points are also written if they exist. 
```

Comment:
you might want to mention that materials are not written out.

Reply:
ok i will re-add the mtl code again

Reply:
i am trying to finish this PR, by calling the getOutputData(1) i am getting an error

```js
Uncaught TypeError: model.output[count].isDeleted is not a function
    at publicAPI.shouldUpdate (macros.js:853:55)
    at Object.getOutputData (macros.js:839:19)
```

in both outputs i am returning strings

```js
const objContent = writer.getOutputData(0);
const mtlContent = writer.getOutputData(1);
```

Displaying the content of `model.output[count]`, i can see it's returned string hence the error, so i am not sure if it's a bug or not.

![image](https://github.com/user-attachments/assets/183ecdaf-e7d1-4814-a4b2-9e357bba9443)

Reply:
Did you check vtkMoleculeToRepresentation that produces 2 outputs ?

Reply:
Yes but it does produce two polydatas that's why it works, here i am producing two strings to be returned

Reply:
I see, I think I got confused at some point. A writer is not really meant to have any output.

`getOutputData()` is supposed to create a `vtkDataSet`, in your case `getOutput()` should be enough.

Reply:
i don't think `getOutput()` exist in vtk.js